common/src/main/java/org/tango/network/NetworkUtils.java

```diff
@@ -0,0 +1,94 @@
+package org.tango.network;
+
+import com.google.common.collect.Collections2;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
```

Comment:
I suggest remove Guava if possible. Using mix of Guava and Java 8 is confusing. Most Guava functions have analogues in Java 8.